tracer/src/Datadog.Trace/Ci/CIVisibility.cs

```diff
@@ -418,5 +441,37 @@ private static async Task GetIntelligentTestRunnerSkippableTestsAsync()
                 Log.Error(ex, "ITR: Error getting skippeable tests.");
             }
         }
+
+        private static bool IsEventPlatformProxySupportedByAgent(IDiscoveryService discoveryService)
+        {
+            var eventPlatformProxyEnabled = false;
+            AgentConfiguration? agentConfiguration = null;
+            ManualResetEventSlim manualResetEventSlim = new(false);
+            LifetimeManager.Instance.AddShutdownTask(() => manualResetEventSlim.Set());
+
+            Log.Debug("Waiting for agent configuration...");
+            discoveryService.SubscribeToChanges(CallBack);
+            void CallBack(AgentConfiguration aConfiguration)
+            {
+                agentConfiguration = aConfiguration;
```

Comment:
Asynchronous programming question...I noticed a `Volatile.Read` below, so do we need to do a `Volatile.Write` here?

Reply:
We can add it but, TBH I don't think is necessary in this case.  No other write or read operation comes after this in this Thread and the `ManualResetEventSlim.Set` already puts a memory barrier (https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Threading/ManualResetEventSlim.cs,282).